source/opt/aggressive_dead_code_elim_pass.cpp

```diff
@@ -321,7 +321,7 @@ bool AggressiveDCEPass::AggressiveDCE(ir::Function* func) {
   for (auto bi = structuredOrder.begin(); bi != structuredOrder.end(); ++bi) {
     for (auto ii = (*bi)->begin(); ii != (*bi)->end(); ++ii) {
       if (IsLive(&*ii)) continue;
-      if (IsBranch(ii->opcode()) &&
+      if (ii->IsBranch() &&
```

Comment:
You've extended the functionality of this pass. You should test it with an OpSwitch.

Reply:
Added a new test, but it fails because ADCE does not seem to handle OpSwitch at all (#1021)